### PR TITLE
ASM-5562 vCenter Puppet module changes for Phase 1 vSAN support for RA HCL

### PR DESCRIPTION
**Pull request by sushilrai**
- Support added for enabling VSAN on a VMKernel portgroup. API for setting VSAN property works in cluster context rather than at the host context
- First VMKernel interface needs to be created and after adding the hosts to the cluster and enabling VSAN, we need to updated VSAN enabled property on the interface

### CODE REVIEW

**Comment by sushilrai** [comment]
Few more resources needs to be added, but those will be independent. review and merge PR, will be adding new PR with other changes.

File: tests/vc_vsan_network.pp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2013 VMware, Inc.
```
Comment (gavin-scott) [inline tests/vc_vsan_network.pp]: Probably not right ;-)

Comment (gavin-scott) [inline tests/vc_vsan_network.pp]: Please update copyright.

**Comment by gavin-scott** [comment]
@asaenz can you review?

File: lib/puppet/provider/vc_vsan_network/default.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
+provider_path = Pathname.new(__FILE__).parent.parent
+require File.join(provider_path, 'vcenter')
+require 'rbvmomi'
+
+Puppet::Type.type(:vc_vsan_network).provide(:vv_vsan, :parent => Puppet::Provider::Vcenter) do
```
Comment (asaenz) [inline lib/puppet/provider/vc_vsan_network/default.rb]: Should this be "vc_vsan", instead "vv_vsan"?

Comment (gavin-scott) [inline lib/puppet/provider/vc_vsan_network/default.rb]: This should be fixed.

**Pull request merged**